ios/browser/api/history/brave_history_api.mm

```diff
@@ -109,6 +109,127 @@ - (NSDate*)dateAdded {
 }
 @end
 
+#pragma mark - IOSHistorySearchOptions
+
+@interface IOSHistorySearchOptions () {
+  int max_count_;
+  bool host_only_;
+  base::Time begin_time_;
+  base::Time end_time_;
+  history::QueryOptions::DuplicateHandling duplicate_policy_;
+}
+@end
+
+@implementation IOSHistorySearchOptions
+
+- (instancetype)init {
+  return [self initWithMaxCount:0
+                       hostOnly:NO
+              duplicateHandling:HistoryDuplicateHandlingIOSRemoveAll
+                      beginDate:nil
+                        endDate:nil];
+}
+
+- (instancetype)initWithMaxCount:(NSUInteger)maxCount
+               duplicateHandling:
+                   (HistoryDuplicateHandlingIOS)duplicateHandling {
+  return [self initWithMaxCount:maxCount
+                       hostOnly:NO
+              duplicateHandling:duplicateHandling
+                      beginDate:nil
+                        endDate:nil];
+}
+
+- (instancetype)initWithMaxCount:(NSUInteger)maxCount
+                        hostOnly:(BOOL)hostOnly
+               duplicateHandling:(HistoryDuplicateHandlingIOS)duplicateHandling
+                       beginDate:(nullable NSDate*)beginDate
+                         endDate:(nullable NSDate*)endDate {
+  if ((self = [super init])) {
+    [self setMaxCount:maxCount];
+    [self setHostOnly:hostOnly];
+    [self setDuplicateHandling:duplicateHandling];
+    if (beginDate) {
+      [self setBeginDate:beginDate];
+    }
+    if (endDate) {
+      [self setEndDate:endDate];
+    }
+  }
+  return self;
+}
+
+- (void)setMaxCount:(NSUInteger)maxCount {
+  max_count_ = static_cast<int>(maxCount);
+}
+
+- (NSUInteger)maxCount {
+  return static_cast<NSUInteger>(max_count_);
+}
+
+- (void)setHostOnly:(BOOL)hostOnly {
+  host_only_ = hostOnly;
```

Comment:
Should not have a `host_only_` variable. All of these setters can be removed.

Reply:
Again some rebase problems